### PR TITLE
atomic PrewriteMaxBackoff

### LINKED ISSUE #487: DARA RACE in the PrewriteMaxBackoff
when to run the test case in the TiDB, you will find the data race.
```
==================
WARNING: DATA RACE
Read at 0x00000783c8c8 by goroutine 141:
  github.com/tikv/client-go/v2/txnkv/transaction.(*twoPhaseCommitter).execute()
      /home/prow/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.1-0.20220406091203-f73ec0e675f4/txnkv/transaction/2pc.go:1372 +0x2c8
  github.com/tikv/client-go/v2/txnkv/transaction.(*KVTxn).Commit()
      /home/prow/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.1-0.20220406091203-f73ec0e675f4/txnkv/transaction/txn.go:436 +0xc7b
  github.com/pingcap/tidb/store/driver/txn.(*tikvTxn).Commit()
      /go/tidb/store/driver/txn/txn_driver.go:77 +0x53
  github.com/pingcap/tidb/session.(*LazyTxn).Commit()
      /go/tidb/session/txn.go:365 +0x76c
  github.com/pingcap/tidb/session.(*session).commitTxnWithTemporaryData()
      /go/tidb/session/session.go:730 +0x2c3
  github.com/pingcap/tidb/session.(*session).doCommit()
      /go/tidb/session/session.go:620 +0x145a
  github.com/pingcap/tidb/session.(*session).doCommitWithRetry()
      /go/tidb/session/session.go:861 +0x3a4
  github.com/pingcap/tidb/session.(*session).CommitTxn()
      /go/tidb/session/session.go:927 +0x2c4
  github.com/pingcap/tidb/session.autoCommitAfterStmt()
      /go/tidb/session/tidb.go:280 +0x3c7
  github.com/pingcap/tidb/session.finishStmt()
      /go/tidb/session/tidb.go:246 +0x3ab
  github.com/pingcap/tidb/session.runStmt()
      /go/tidb/session/session.go:2067 +0x91c
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      /go/tidb/session/session.go:1932 +0xdda
  github.com/pingcap/tidb/session.(*session).ExecuteInternal()
      /go/tidb/session/session.go:1465 +0x464
  github.com/pingcap/tidb/ddl/util.UpdateDeleteRange()
      /go/tidb/ddl/util/util.go:162 +0x50f
  github.com/pingcap/tidb/ddl.(*delRange).doTask()
      /go/tidb/ddl/delete_range.go:232 +0x2b3
  github.com/pingcap/tidb/ddl.(*delRange).doDelRangeWork()
      /go/tidb/ddl/delete_range.go:169 +0x64b
  github.com/pingcap/tidb/ddl.(*delRange).startEmulator()
      /go/tidb/ddl/delete_range.go:148 +0x1c8
  github.com/pingcap/tidb/ddl.(*delRange).start.func1()
      /go/tidb/ddl/delete_range.go:124 +0x39
Previous write at 0x00000783c8c8 by goroutine 19:
  github.com/pingcap/tidb/tests/pessimistictest.createMockStoreAndSetup.func2()
      /go/tidb/tests/pessimistictest/pessimistic_test.go:159 +0x51
  runtime.deferreturn()
      /usr/local/go/src/runtime/panic.go:436 +0x32
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
Goroutine 141 (running) created at:
  github.com/pingcap/tidb/ddl.(*delRange).start()
      /go/tidb/ddl/delete_range.go:124 +0xb6
  github.com/pingcap/tidb/ddl.(*ddl).newDeleteRangeManager()
      /go/tidb/ddl/ddl.go:373 +0x245
  github.com/pingcap/tidb/ddl.(*ddl).Start()
      /go/tidb/ddl/ddl.go:395 +0x4eb
  github.com/pingcap/tidb/domain.(*Domain).Init()
      /go/tidb/domain/domain.go:861 +0x11d6
  github.com/pingcap/tidb/session.(*domainMap).Get.func1()
      /go/tidb/session/tidb.go:85 +0x664
  github.com/pingcap/tidb/util.RunWithRetry()
      /go/tidb/util/misc.go:64 +0x93
  github.com/pingcap/tidb/session.(*domainMap).Get()
      /go/tidb/session/tidb.go:73 +0x319
  github.com/pingcap/tidb/session.createSessionWithOpt()
      /go/tidb/session/session.go:2983 +0x75
  github.com/pingcap/tidb/session.createSession()
      /go/tidb/session/session.go:2979 +0x84
  github.com/pingcap/tidb/session.createSessions()
      /go/tidb/session/session.go:2968 +0x6e
  github.com/pingcap/tidb/session.BootstrapSession()
      /go/tidb/session/session.go:2845 +0x297
  github.com/pingcap/tidb/tests/pessimistictest.createMockStoreAndSetup()
      /go/tidb/tests/pessimistictest/pessimistic_test.go:154 +0x288
  github.com/pingcap/tidb/tests/pessimistictest.TestInnodbLockWaitTimeout()
      /go/tidb/tests/pessimistictest/pessimistic_test.go:934 +0x51
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
Goroutine 19 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1839 +0x99
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1837 +0x7e4
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1719 +0xa71
  go.uber.org/goleak.VerifyTestMain()
      /home/prow/go/pkg/mod/go.uber.org/goleak@v1.1.12/testmain.go:53 +0x59
  github.com/pingcap/tidb/tests/pessimistictest.TestMain()
      /go/tidb/tests/pessimistictest/pessimistic_test.go:85 +0xc04
  main.main()
      _testmain.go:165 +0x317
================== 
```

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

close #487

### CODE REVIEW

File: txnkv/transaction/2pc.go [hunk 1]
```diff
@@ -82,7 +83,7 @@ var (
 
 var (
 	// PrewriteMaxBackoff is max sleep time of the `pre-write` command.
-	PrewriteMaxBackoff = 40000
+	PrewriteMaxBackoff = atomicutil.NewUint32(40000)
```
Comment (sticnarf) [inline txnkv/transaction/2pc.go:86]: `CommitMaxBackoff` is a uint64. What about also use uint64 here to be more consistent?

Comment (hawkingrei) [reply txnkv/transaction/2pc.go]: OK, have updated.

**Review by sticnarf — APPROVED** [review]
LGTM. Please also update the client-go in TiDB asap to resolve the API breakage.

**Comment by hawkingrei** [comment]
> LGTM. Please also update the client-go in TiDB asap to resolve the API breakage.

OK, I will.

**Pull request merged**